### ISSUE TITLE
Integer arguments to integrate 1d invoking wrong version

### ISSUE THREAD

**Issue by bbbales2** [stan-dev/math#935]
#### Summary:

Trying to call the prim version of integrate_1d with integer arguments invokes the wrong version of the function.

#### Description:

Because the reverse mode integrate_1d is most templated, if the limits are not doubles (but are not vars and theta is not vars), the compiler tries to use the reverse mode version of integrate_1d anyway.

#### Reproducible Steps:

Add the code:
```
TEST(StanMath_integrate_1d, test_integer_arguments) {
  double out = stan::math::integrate_1d(f2{}, 0, 1, std::vector<double>(),
                                        {}, {}, msgs, 1e-6);
}
```

To test/unit/math/prim/arr/functor/integrate_1d_test.cpp

And run ./runTests.py test/unit/math/prim/arr/functor/integrate_1d_test.cpp

#### Current Output:

Fails at compile:
```
test/unit/math/prim/arr/functor/integrate_1d_test.cpp:288:10: error: no viable conversion from 'stan::math::var' to 'double'
  double out = stan::math::integrate_1d(f2{}, 0, 1, std::vector<double>(),
```

#### Expected Output:

Passes

#### Current Version:
v2.17.0

**Comment by andrjohns** [comment]
Closed via #936

**Comment by bob-carpenter** [comment]
I'm guessing this may be a problem for a lot of our code.  It would be nice if `int` to `double` were a cheaper promotion than `int` to `var`, but it's not.  

I'm guessing *a lot* of our functions suffer this problem.  I'm going to open a discussion on Discourse.
